Platform/TigerlakeBoardPkg/BoardConfig.py

```diff
@@ -195,13 +194,34 @@ def __init__(self, *args, **kwargs):
         self.LOADER_RSVD_MEM_SIZE = 0x500000
 
         self.CFG_DATABASE_SIZE    = self.CFGDATA_SIZE
+        self._generated_cfg_file_prefix = 'Autogen_'
 
         # _CFGDATA_INT_FILE - Internal cfg data is generally used for internal boards like MRBs, RVPs etc.
         # _CFGDATA_EXT_FILE - External cfg data is for the customer boards to populate new data on top of the internal defaults.
-        # Cfg data dlt files for nternal boards could also put into external cfg data if want to update cfg data for these platforms
+        # Cfg data dlt files for internal boards could also put into external cfg data if want to update cfg data for these platforms
         # for test purpose. Based on the platform id, relevant data is populated for each platform.
         self._CFGDATA_INT_FILE = []
-        self._CFGDATA_EXT_FILE = ['CfgData_Ext_Dummy.dlt', 'CfgData_Int_Tglu_Ddr4.dlt', 'CfgData_Int_Tglu_DdrLp4.dlt']
+        self._CFGDATA_EXT_FILE = [self._generated_cfg_file_prefix + 'CfgData_Int_Tglu_Ddr4.dlt', self._generated_cfg_file_prefix  + 'CfgData_Int_Tglu_DdrLp4.dlt']
+
+    def PlatformBuildHook (self, build, phase):
+        if phase == 'pre-build:before':
+            # create build folder if not exist
+            if not os.path.exists(build._fv_dir):
+                os.makedirs(build._fv_dir)
+
+            # Generate the dlt files based on feature
+            brd_cfg_src_dir = os.path.join(os.environ['PLT_SOURCE'], 'Platform', self.BOARD_PKG_NAME, 'CfgData')
+            for dlt_file in self._CFGDATA_EXT_FILE:
+                cfg_dlt_file  = os.path.join(brd_cfg_src_dir, dlt_file[len (self._generated_cfg_file_prefix):])
+                lines         = open (cfg_dlt_file).read()
+
+                # Enable TCC in dlt file
+                if self.ENABLE_TCC:
+                    lines += open (os.path.join(brd_cfg_src_dir, 'CfgData_Tcc_Feature.dlt')).read()
+
+                # Write to generated final dlt file
+                output_cfg_dlt_file = os.path.join(build._fv_dir, dlt_file)
+                open(output_cfg_dlt_file, 'w').write(lines)

```

Comment:
where do you delete the generated files ?   
If you don't do it,  the generated dlt files will not be updated even when the source files are modified.

Reply:
It is not necessary to delete the generated one in build folder since BoardConfig.py will re-generate it from source in every build.

Reply:
OK.  But you might want to check if the generated CfgDataStitch.py in Build folder still works as before.
The python file will use the DELTA files.

Reply:
yes.  the generated CfgDataStitch.py works with autogenerated dlt files.

Reply:
OK.